src/main/java/de/tum/cit/aet/artemis/programming/web/ProgrammingExerciseResource.java

```diff
@@ -531,6 +532,23 @@ public ResponseEntity<ProgrammingExercise> getProgrammingExerciseWithTemplateAnd
         return ResponseEntity.ok(programmingExercise);
     }
 
+    /**
+     * GET /programming-exercises : Queries a programming exercise by its project key.
+     *
+     * @return the ProgrammingExercise of this project key in an ResponseEntity or 404 Not Found if no exercise exists
+     */
+    @GetMapping("programming-exercises/project-key/{projectKey}")
+    @EnforceAtLeastStudent
+    public ResponseEntity<ProgrammingExercise> getExerciseByProjectKey(@PathVariable String projectKey) {
+        try {
```

Comment:
You can just re-throw the exception.

Reply:
but wouldnt rethrowing trigger a internal server error response?